spec/dynamoid/transaction_write/upsert_spec.rb

```diff
@@ -94,5 +94,65 @@
       expect(obj3_found.created_at).to be_nil
       expect(obj3_found.updated_at.to_f).to be_within(1.seconds).of Time.now.to_f
     end
+
+    context 'adds' do
+      it 'to nil' do
+        klass.create_table
+        obj3_id = SecureRandom.uuid
+        described_class.execute do |txn|
+          txn.upsert klass, { id: obj3_id, name: 'threethree' } do |u|
+            u.add(record_count: 5)
+          end
+        end
+        obj3_found = klass.find(obj3_id)
+        expect(obj3_found.id).to eql(obj3_id)
+        expect(obj3_found.name).to eql('threethree')
+        expect(obj3_found.record_count).to eql(5)
+      end
+
+      it 'to a value' do
+        klass.create_table
+        obj3_id = SecureRandom.uuid
+        klass.create!(id: obj3_id, name: 'three', record_count: 50)
+        described_class.execute do |txn|
+          txn.upsert klass, { id: obj3_id, name: 'threethree' } do |u|
+            u.add(record_count: 5)
+          end
+        end
+        obj3_found = klass.find(obj3_id)
+        expect(obj3_found.id).to eql(obj3_id)
+        expect(obj3_found.name).to eql('threethree')
+        expect(obj3_found.record_count).to eql(55)
+      end
+
+      it 'to nil set' do
+        klass.create_table
+        obj3_id = SecureRandom.uuid
+        described_class.execute do |txn|
+          txn.upsert klass, { id: obj3_id, name: 'threethree' } do |u|
+            u.add(favorite_names: %w[adam bob])
+          end
+        end
+        obj3_found = klass.find(obj3_id)
+        expect(obj3_found.id).to eql(obj3_id)
+        expect(obj3_found.name).to eql('threethree')
+        expect(obj3_found.favorite_names).to eql(Set.new(%w[adam bob]))
+      end
+
+      it 'to existing set' do
+        klass.create_table
+        obj3_id = SecureRandom.uuid
+        klass.create!(id: obj3_id, name: 'three', favorite_numbers: [1, 2])
+        described_class.execute do |txn|
+          txn.upsert klass, { id: obj3_id, name: 'threethree' } do |u|
+            u.add(favorite_numbers: [2, 3])
+          end
+        end
+        obj3_found = klass.find(obj3_id)
+        expect(obj3_found.id).to eql(obj3_id)
+        expect(obj3_found.name).to eql('threethree')
+        expect(obj3_found.favorite_numbers).to eql(Set[1, 2, 3])
+      end
+    end
```

Comment:
If the `.update` method doesn't except a block - this shouldn't be tested as well.

Reply:
I think you meant `.upsert`. This is useful functionality that hopefully makes it back to the docs and tests soon. I removed these tests.

Reply:
Right 👍